src/server/sched_func.c

```diff
@@ -331,13 +381,28 @@ int
 poke_scheduler(attribute *pattr, void *pobj, int actmode)
 {
 	if (pobj == &server || pobj == dflt_scheduler) {
+		if (pobj == &server) {
+			/* set this attribute on main scheduler */
+			if (dflt_scheduler) {
+				dflt_scheduler->sch_attr[SCHED_ATR_scheduling].at_val.at_long = pattr->at_val.at_long;
+				dflt_scheduler->sch_attr[SCHED_ATR_scheduling].at_flags |=
+						ATR_VFLAG_SET | ATR_VFLAG_MODIFY | ATR_VFLAG_MODCACHE;
+				(void)sched_save_db(dflt_scheduler, SVR_SAVE_FULL);
+			}
+		} else {
+			server.sv_attr[SRV_ATR_scheduling].at_val.at_long = pattr->at_val.at_long;
```

Comment:
when do we reach this else case? If we come here when we are trying to set scheduling on a scheduler object then why do we set server's attribute here? I'm confused.

Reply:
The else block is hit when admin tries to set scheduling on default scheduler. In this case we also have to make sure that server's scheduling attribute is set to the same value as that of default scheduler.  This is to make sure that data is consistent between default scheduler vs server. Also making sure that vice versa case is also consistent.